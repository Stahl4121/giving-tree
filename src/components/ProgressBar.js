import React, { useEffect } from 'react';
import Text from './Text.js';
import LinearProgress from '@material-ui/core/LinearProgress';

export default function ProgressBar(props) {
  const [raised, setRaised] = React.useState(props.raised);
  const [goal, setGoal] = React.useState(props.goal);
  const [progress, setProgress] = React.useState(100 * Number.parseFloat(raised) / Number.parseFloat(goal))

  useEffect(() => { setRaised(props.raised); }, [props.raised]);

  useEffect(() => { setGoal(props.goal); }, [props.goal]);

  useEffect(() => { setProgress(100 * Number.parseFloat(raised) / Number.parseFloat(goal)); }, [props.goal, props.raised]);

  return (
    <div>
      <Text type='card-subheading' text={'$' + props.raised + ' raised out of $' + props.goal} />
      <LinearProgress value={progress} variant="determinate" />
    </div>
  );
}
