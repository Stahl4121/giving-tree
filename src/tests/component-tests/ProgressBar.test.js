import React from "react";
import { shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import ProgressBar from "../../components/ProgressBar.js";

describe('Component: ProgressBar', () => {
  var enzyme = require('enzyme');
  enzyme.configure({ adapter: new Adapter() });

  it('Renders correctly', () => {
    const onSelect = jest.fn();
    const wrap = enzyme.mount(<ProgressBar parentCallback={onSelect} />)
    expect(wrap).toMatchSnapshot();
  });

  it('Sort by deadline', () => {
    const onSelect = jest.fn();
    const wrap = enzyme.mount(<ProgressBar parentCallback={onSelect} />)
    const select = wrap.find('#select-sort').at(0);
    select.props().onChange({ target: { value: "deadline"} });
    expect(onSelect.mock.calls.length).toBe(1);
    expect(onSelect.mock.calls[0][0]).toBe("deadline");
  });
  
})