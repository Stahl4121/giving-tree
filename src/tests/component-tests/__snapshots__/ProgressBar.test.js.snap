// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: ProgressBar Renders correctly 1`] = `ReactWrapper {}`;
